frontend/awx/access/credentials/CredentialForm.tsx

```diff
@@ -818,70 +862,133 @@ function CredentialTextInput({
     return helperText;
   };
 
-  const renderFieldValue = useCallback(
-    (field: CredentialInputField): string => {
-      let placeholder = '';
-      accumulatedPluginValues.forEach((cp) => {
-        if (cp.input_field_name === field.id && sourceCredential) {
-          placeholder = t(`Value is managed by ${sourceCredential.kind}: ${sourceCredential.name}`);
-        }
-      });
-      return placeholder;
-    },
-    [accumulatedPluginValues, sourceCredential, t]
-  );
+  const clearField = () => {
+    setValue(field.id, '', { shouldDirty: false });
+    setShouldHideField(false);
+  };
+
+  const revertInitialValue = () => {
+    setValue(field.id, fieldInitialValue, { shouldDirty: true });
+    setAccumulatedPluginValues?.(
+      accumulatedPluginValues.filter((cp) => cp.input_field_name !== field.id)
+    );
+    setShouldHideField(true);
+  };
 
   useEffect(() => {
-    // mark field as managed by a credential plugin
-    if (accumulatedPluginValues.some((cp) => cp.input_field_name === field.id)) {
-      setValue(field.id, renderFieldValue(field), { shouldDirty: true });
+    if (shouldHideField && isSecretField && isInitialValueEncrypted && !isPromptOnLaunchChecked) {
+      setValue(field.id, t('ENCRYPTED'), { shouldDirty: false });
```

Comment:
This actually updates the stored password value that the API returns (`$encrypted$`) to `ENCRYPTED` and sends it as the new string to the API, which we don't want. We should set it to `initialValues[field.id]` so that it doesn't change the value.